### PR TITLE
[PIP]Implement Robobrain War Mind

### PR DESCRIPTION
**Pull request by justinjohnson14**
Sorry for creating a second one I'm still new to git and got scared lol.

### CODE REVIEW

File: Mage.Sets/src/mage/cards/r/RobobrainWarMind.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        super(ownerId, setInfo, new CardType[]{CardType.ARTIFACT, CardType.CREATURE}, "{3}{U}");
+        
+        this.subtype.add(SubType.ROBOT);
+        this.power = new MageInt(0);
+        this.toughness = new MageInt(5);
+
+        // Robobrain War Mind's power is equal to the number of cards in your hand.
+        this.addAbility(new SimpleStaticAbility(Zone.ALL, new SetBasePowerSourceEffect(CardsInControllerHandCount.instance)));
+        //  Robobrain War Mind enters the battlefield, you get an amount of {E} equal to the number of artifact creatures you control.
+        this.addAbility(new EntersBattlefieldTriggeredAbility(new GetEnergyCountersControllerEffect(ArtifactYouControlCount.instance)));
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/r/RobobrainWarMind.java:38]: wrong dynamic value, needs to be artifact creatures, not artifacts, you control

use PermanentsOnBattlefieldCount with appropriate filter, both declared static

**Review by xenohedron — CHANGES_REQUESTED** [review]

**Comment by justinjohnson14** [comment]
I give up i apparently don't know what I am doing.

**Comment by ssk97** [comment]
> I give up i apparently don't know what I am doing.

You want to create a new branch from master each time you want to make a new Pull Request. You can always check to see which branch you're currently in.

**Comment by justinjohnson14** [comment]
If this pull request gets closed. Can I commit changes to main without it getting attached to this pull request again? I plan to make named branches for card implementations now but just wondering for future reference.

File: Mage.Sets/src/mage/cards/r/RobobrainWarMind.java [hunk 2]
```diff
@@ -50,3 +68,30 @@ public RobobrainWarMind copy() {
         return new RobobrainWarMind(this);
     }
 }
+class RobobrainWarMindEffect extends OneShotEffect {
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/r/RobobrainWarMind.java:71]: Don't make a new effect class. Use `GetEnergyCountersControllerEffect`

File: Mage.Sets/src/mage/cards/r/RobobrainWarMind.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        this.toughness = new MageInt(5);
+
+        // Robobrain War Mind's power is equal to the number of cards in your hand.
+        this.addAbility(new SimpleStaticAbility(Zone.ALL, new SetBasePowerSourceEffect(CardsInControllerHandCount.instance)));
+
+        // When Robobrain War Mind enters the battlefield, you get an amount of {E} equal to the number of artifact creatures you control.
+        this.addAbility(new EntersBattlefieldTriggeredAbility(new GetEnergyCountersControllerEffect(new PermanentsOnBattlefieldCount(filter))));
+
+        // Whenever Robobrain War Mind attacks, you may pay {E}{E}{E}. If you do, draw a card.
+        this.addAbility(new AttacksTriggeredAbility(new DoIfCostPaid(new DrawCardTargetEffect(1, false), new PayEnergyCost(3))));
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/r/RobobrainWarMind.java:62]: use `DrawCardSourceControllerEffect`, this isn't targeted.

**Comment by xenohedron** [comment]
> If this pull request gets closed. Can I commit changes to main without it getting attached to this pull request again? I plan to make named branches for card implementations now but just wondering for future reference.

Once the PR is closed/merged it'll be fine. What I recommend is after this gets merged, to reset the master branch of your fork to magefree:master.

**Review by xenohedron — APPROVED** [review]

**Comment by xenohedron** [comment]
thanks for the contribution

**Pull request merged**